compiler/rustc_infer/src/errors/mod.rs

```diff
@@ -1157,3 +1145,388 @@ pub struct OpaqueCapturesLifetime<'tcx> {
     pub opaque_ty_span: Span,
     pub opaque_ty: Ty<'tcx>,
 }
+
+pub struct DiagArg<T>(pub T);
```

Comment:
Why is this type necessary? Could [`rustc_errors::diagnostic_impls::DiagnosticArgFromDisplay`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_errors/diagnostic_impls/struct.DiagnosticArgFromDisplay.html#) be used instead?

Reply:
Resolved by implementing `IntoDiagnosticArg` for `Binder`